File: fixtures/scale/r/large/core/report.r
# Post-run report over the stored results.

report_line <- function(row) {
  paste0("  ", gsub('","', "  ", row, fixed = TRUE))
}

report_summary <- function() {
  path <- results_path()
  if (!file.exists(path)) {
    log_warn("no results at", path)
    return(invisible(NULL))
  }
  cat(sprintf("results from %s:\n", path))
  for (row in readLines(path)) {
    cat(report_line(row), "\n")
  }
}

File: fixtures/scale/julia/large/src/core/Report.jl
module BenchReport

using ..BenchStore
using ..BenchLog

export report_summary

function report_line(row::String)
    return "  " * replace(row, "\",\"" => "  ")
end

function report_summary()
    path = BenchStore.results_path()
    if !isfile(path)
        BenchLog.log_warn("no results at", path)
        return
    end
    println("results from ", path, ":")
    for row in eachline(path)
        println(report_line(row))
    end
end

end
